WIP: Finish can keep guests running

lukaszachy opened the pull request:
Solves #208

Early feedback please. Need to solve
- correct way of printing run id
- how to detect proper way to connect (e.g. no point for localhost)
- better guest.name handling?

Use as
```
$ tmt run -a finish --keep
/var/tmp/tmt/run-060

/tests/ABC
    discover:
     ....
    finish
        Guest will keep running
        Run `tmt run --id /var/tmp/tmt/run-060 finish` later to finish properly
        guest: default
        connect: ssh root@A.B.C.D -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -i /path/to/key
```

psss commented on tmt/steps/finish/__init__.py:
```diff
@@ -29,7 +29,16 @@ def go(self):
+            if self.opt('keep'):
+                # Print details about connection
+                self.info("guest", guest.name, color='magenta', shift=1)
+                self.info('connect', "ssh {} {}".format(guest._ssh_guest(), guest._ssh_options(True)), 'green', shift=1)
```
I think this part should go to the `Guest` class. We could introduce a new `keep()` method which would serve this information and allow inheriting classes to override if necessary. This will be useful for example for the `podman` plugin which can provide its own hint how to connect to the container.

psss commented:
The dynamic `finish` steps is ready for review in #230.

lukaszachy commented:
Rebased and changed on top of dynamic-finish. I like the 'keep' method of guests, maybe it can be somehow shared with login subcommand (#149).

One more issue though - we have 'status' of step but it looks we need some way of marking partial progress. E.g. on `tmt run -a finish --keep` I would expect to run all finish step except returning the guests. So when followed with `tmt run --id XX finish` it will just remove the guests.

Or should it run (if defined) finish steps only if it would remove the guests (when no --keep is used)?

psss commented:
> One more issue though - we have 'status' of step but it looks we need some way of marking partial progress.

I see. This could be probably also achieved by storing an additional information in the step data. If possible I would like to prevent introducing an additional status unless really necessary.

> Or should it run (if defined) finish steps only if it would remove the guests (when no --keep is used)?

If I understand it correctly this would be similar (or even identical) to the following?

    tmt run --until report
    tmt run --last finish

I think, for now it could be sufficient (until there are more complex finishing tasks).

lukaszachy commented:
With login, --until and similar options I think this PR lost it's purpose.
